pkg/controller/persistentvolume/controller_base.go

```diff
@@ -139,14 +140,17 @@ func (ctrl *PersistentVolumeController) initializeCaches(volumeSource, claimSour
 		glog.Errorf("PersistentVolumeController can't initialize caches: %v", err)
 		return
 	}
-	volumeList, ok := volumeListObj.(*api.List)
-	if !ok {
-		glog.Errorf("PersistentVolumeController can't initialize caches, expected list of volumes, got: %+v", volumeListObj)
-		return
-	}
-	for _, volume := range volumeList.Items {
+	volumeListVal := reflect.Indirect(reflect.ValueOf(volumeListObj))
```

Comment:
Please add a comment why `reflect` is needed as both `List` and `PersistentVolumeList` need to be processed.